PerimeterXModule/Internals/ReverseProxy.cs

```diff
@@ -0,0 +1,240 @@
+		 */
+		private bool ProcessRequest(HttpContext context, string serverUrl, string uri)
+		{
+			try
+			{
+				// Set headers
+				context.Request.Headers.Add(PxConstants.ENFORCER_TRUE_IP_HEADER, PxCommonUtils.GetRequestIP(context, PxConfig));
+				context.Request.Headers.Add(PxConstants.FIRST_PARTY_HEADER, PxConstants.FIRST_PARTY_VALUE);
+
+				// Create a connexion to the Remote Server to redirect all requests
+				RemoteServer server = new RemoteServer(context, serverUrl, uri);
+
+				// Create a request with same data in navigator request
+				HttpWebRequest request = server.GetRequest();
+
+				// Send the request to the remote server and return the response
+				HttpWebResponse response = server.GetResponse(request);
+				if (response == null || !response.StatusCode.Equals(HttpStatusCode.OK))
+				{
+					Debug.WriteLine("ReverseProxy responeded with none 200 status", PxConstants.LOG_CATEGORY);
+					Debug.WriteLineIf(!(response == null), "Response status {0}", PxConstants.LOG_CATEGORY);
+					return false;
+				}
+
+				byte[] responseData = server.GetResponseStreamBytes(response);
+
+				// Send the response to client
+				context.Response.Headers.Add(response.Headers);
+				context.Response.ContentEncoding = Encoding.UTF8;
+				context.Response.ContentType = response.ContentType;
+				context.Response.OutputStream.Write(responseData, 0, responseData.Length);
+
+				// Handle cookies to navigator
+				server.SetContextCookies(response);
+
+				// Close streams
+				response.Close();
+				return true;
+			}
+			catch (Exception e)
+			{
+				Debug.WriteLine("Unexpected error while processing reverse request: " + e.Message, PxConstants.LOG_CATEGORY);
+				return false;
+			}
+
+		}
+		/**
+		 * <summary>
+		 * Forward requests for PerimeterX client
+		 * </summary>
+		 * <param name="context">HttpContext</param>
+		 */
+		public void ReversePxClient(HttpContext context)
+		{
+			Debug.WriteLine("Fetching Client", PxConstants.LOG_CATEGORY);
+			string contentType = CONTENT_TYPE_JAVASCRIPT;
+			string defaultResponse = DEFAULT_CLIENT_VALUE;
+			if (!PxConfig.FirstPartyEnabled)
+			{
+				Debug.WriteLine("First party is disabled, rendering default response", PxConstants.LOG_CATEGORY);
+				RenderPredefinedResponse(context, contentType, defaultResponse);
+			}
+			string uri = string.Format("/{0}/main.min.js", PxConfig.AppId);
+			bool success = ProcessRequest(context, PxConfig.ClientHostUrl, uri);
+
+			if (!success)
+			{
+				Debug.WriteLine("Redirect XHR returned bad status, rendering default response", PxConstants.LOG_CATEGORY);
+				RenderPredefinedResponse(context, contentType, defaultResponse);
+			}
+
+		}
+
+		/**
+		 * <summary>
+		 * Forward any sensor activities back to PerimeterX servers
+		 * </summary>
+		 * <param name="context">HttpContext</param>
+		 */
+		public void ReversePxXhr(HttpContext context)
+		{
+			string defaultResponse;
+			string contentType;
+
+			if (context.Request.Url.AbsolutePath.EndsWith(".gif"))
```

Comment:
I assume that this also works for `/vRfnOj4y/xhr/api/v1/collector/noScript.gif?appId=PXvRfnOj4y` (such that the absolute path does end with .gif and the rest are stored as query params)?

Reply:
Yes